tests/core/fs/test_bootstrap.py

```diff
@@ -33,13 +41,17 @@ async def test_user_manifest_access_while_speculative(user_fs_factory, alice):
 
 
 @pytest.mark.trio
-async def test_workspace_manifest_access_while_speculative(user_fs_factory, alice):
+async def test_workspace_manifest_access_while_speculative(
+    user_fs_factory: UserFsFactory,
+    alice: LocalDevice,
+):
     # Speculative workspace occurs when workspace is shared to a new user, or
     # when a device gets it local data removed. We use the latter here (even if
     # it is the less likely of the two) given it is simpler to do in the test.
 
     with freeze_time("2000-01-01"):
         async with user_fs_factory(alice) as user_fs:
+            user_fs: UserFS = user_fs
```

Comment:
maybe add the `type` in `user_fs_factory` `return type`

Reply:
done